aws-redshiftserverless-workgroup/src/main/java/software/amazon/redshiftserverless/workgroup/UpdateHandler.java

```diff
@@ -154,82 +175,14 @@ private TagResourceResponse updateTags(final UpdateTagsRequest awsRequest,
-                                                                                  final Exception exception,
-                                                                                  final ProxyClient<RedshiftServerlessClient> client,
-                                                                                  final ResourceModel model,
-                                                                                  final CallbackContext context) {
-        if (exception instanceof ResourceNotFoundException) {
-            return ProgressEvent.defaultFailureHandler(exception, HandlerErrorCode.NotFound);
-
-        } else if (exception instanceof ValidationException ||
-                exception instanceof TooManyTagsException) {
-            return ProgressEvent.defaultFailureHandler(exception, HandlerErrorCode.InvalidRequest);
-
-        } else if (exception instanceof ThrottlingException) {
-            return ProgressEvent.defaultFailureHandler(exception, HandlerErrorCode.Throttling);
-
-        } else if (exception instanceof InternalServerException) {
-            return ProgressEvent.defaultFailureHandler(exception, HandlerErrorCode.InternalFailure);
-
-        } else {
-            return ProgressEvent.defaultFailureHandler(exception, HandlerErrorCode.GeneralServiceException);
-        }
-    }
-
-    private UpdateWorkgroupResponse updateWorkgroup(final UpdateWorkgroupRequest awsRequest,
-                                                    final ProxyClient<RedshiftServerlessClient> proxyClient) {
-        final int MAX_RETRIES = 5;
-        int retryCount = 0;
-
-        while (true) {
-            try {
-                UpdateWorkgroupResponse awsResponse =
-                        proxyClient.injectCredentialsAndInvokeV2(awsRequest, proxyClient.client()::updateWorkgroup);
-
-                logger.log(String.format("%s has successfully been updated.", ResourceModel.TYPE_NAME));
-
-                return awsResponse;
-            } catch (ConflictException ex) {
-                if (retryCount >= MAX_RETRIES || !isRetriableWorkgroupException(ex)) {
-                    throw ex;
-                }
-
-                logger.log(String.format("Retrying UpdateWorkgroup due to expected ConflictException: " +
-                        "%s. Attempt %d/%d", ex.getMessage(), retryCount + 1, MAX_RETRIES));
-                retryCount++;
-            }
-
-            try {
-                Thread.sleep(10000);
-            } catch (InterruptedException ie) {
-                Thread.currentThread().interrupt(); // Restore the interrupted status
-                throw new RuntimeException("Interrupted during retry wait", ie);
-            }
-        }
-
-    }
-
-    private ProgressEvent<ResourceModel, CallbackContext> updateWorkgroupErrorHandler(final UpdateWorkgroupRequest awsRequest,
+    private ProgressEvent<ResourceModel, CallbackContext> updateWorkgroupErrorHandler(final Object awsRequest,
```

Comment:
keep the type?

Reply:
There are multiple stabilization calls which are using the same error handler so aws request makes sense here. We are logging the exception operation in logs so we should be good here.